### PR TITLE
Chore: Detecting password presence for users registered via social providers

### PR DESCRIPTION
**Pull request by borkopetrovicc**
## Description

Notion card: [Detecting password presence for users registered via social providers](https://www.notion.so/jwplayer/Detecting-password-presence-for-users-registered-via-social-providers-213c5ab6206e4cc59e0c58f8d751cf10)

This PR solves # .

### Steps completed:

According to our definition of done, I have completed the following steps:

- [ ] Acceptance criteria met
- [ ] Unit tests added
- [ ] Docs updated (including config and env variables)
- [ ] Translations added
- [ ] UX tested
- [ ] Browsers / platforms tested
- [ ] Rebased & ready to merge without conflicts
- [ ] Reviewed own code

### CODE REVIEW

File: src/components/Account/Account.tsx
Comment (kmitrovv) [inline src/components/Account/Account.tsx]: This check is inplayer specific and shouldn't be done here in the component. I would consider setting up a flag in the account store for this matter. And what does this user?.metadata?.up mean?

File: src/components/Account/Account.tsx
Comment (kmitrovv) [inline src/components/Account/Account.tsx]: can we replace the conditional rendering here with a standard if statement? would be nice to handle error in case of something happens, so it would be better to wrap it in try catch block.

File: src/components/DeleteAccountModal/DeleteAccountModal.tsx
Comment (kmitrovv) [inline src/components/DeleteAccountModal/DeleteAccountModal.tsx:59]: why is this removed?

File: src/containers/AccountModal/forms/EditPassword.tsx
Comment (kmitrovv) [inline src/containers/AccountModal/forms/EditPassword.tsx]: can we handle the type inline

File: src/components/Account/Account.tsx
Comment (borkopetrovicc) [reply src/components/Account/Account.tsx]: 01d8087341ffc8c1c9bef88c0c40dffa5d178160

File: src/containers/AccountModal/forms/EditPassword.tsx
Comment (borkopetrovicc) [reply src/containers/AccountModal/forms/EditPassword.tsx]: 66f9c2d3a1f608c3acb984248ddd3140419439da

File: src/components/DeleteAccountModal/DeleteAccountModal.tsx
Comment (borkopetrovicc) [reply src/components/DeleteAccountModal/DeleteAccountModal.tsx:59]: Screenshot 2023-10-31 at 14 53 02

Comment (borkopetrovicc) [reply src/components/DeleteAccountModal/DeleteAccountModal.tsx:59]: Because i found bag, that when you go cancel on the modal it gets into infinity loop.

File: src/components/Account/Account.tsx
Comment (borkopetrovicc) [reply src/components/Account/Account.tsx]: it's just a flag for the account so we know if a password has been added it is not necessarily comprehensible to users.

File: src/components/Account/Account.tsx
Comment (kmitrovv) [inline src/components/Account/Account.tsx]: useAccountStore is already defined at line 56

File: src/components/Account/Account.tsx
Comment (kmitrovv) [inline src/components/Account/Account.tsx]: move this below useAccountStore definition:
`const shouldAddPassword = user?.metadata?.register_source !== 'inplayer' && !user?.metadata?.has_password;`

Comment (kmitrovv) [reply src/components/Account/Account.tsx]: But it would be better if this is moved on service level, this way it will probably interfere with Cleeng's logic.

File: src/components/Account/Account.tsx
Comment (kmitrovv) [inline src/components/Account/Account.tsx]: Use an early return so you don't have nested conditions.

File: src/components/EditPasswordForm/EditPasswordForm.tsx
Comment (kmitrovv) [inline src/components/EditPasswordForm/EditPasswordForm.tsx]: missing error handling

File: src/components/WarningAccountModal/WarningAccountModal.tsx
Comment (kmitrovv) [inline src/components/WarningAccountModal/WarningAccountModal.tsx]: Missing error handling

File: src/components/Account/Account.tsx
Comment (ChristiaanScheermeijer) [inline src/components/Account/Account.tsx]: To make the translations extractable:

```suggestion
                label={shouldAddPassword ? t('account.add_password') : t('account.edit_password')}
```

File: src/components/EditPasswordForm/EditPasswordForm.tsx
Comment (ChristiaanScheermeijer) [inline src/components/EditPasswordForm/EditPasswordForm.tsx]:
```suggestion
      <h2 className={styles.title}>{!showOldPasswordField || !showResetTokenField ? t('reset.password_reset') : t('user:account.add_password')}</h2>
```

**Review by ChristiaanScheermeijer — COMMENTED** [review]
LGTM! 

- Have you run `yarn i18next` to extract all translations?
- Perhaps you can rebase/squash the commits to a single commit to make it cleaner. There are some faulty commit messages as well.

File: src/components/Account/Account.tsx
Comment (kmitrovv) [inline src/components/Account/Account.tsx:69]:
```suggestion

// users authenticated with social (register_source: facebook, google, twitter) do not have password by default
```

**Comment by borkopetrovicc** [comment]
> LGTM!
> 
> * Have you run `yarn i18next` to extract all translations?
> * Perhaps you can rebase/squash the commits to a single commit to make it cleaner. There are some faulty commit messages as well.

I have run yarn i18next. Good point for the commits. I will be more carefully for others pr-s.

**Review by kmitrovv — APPROVED** [review]
lgtm

File: src/containers/AccountModal/forms/EditPassword.tsx
Comment (naumovski-filip) [inline src/containers/AccountModal/forms/EditPassword.tsx]:
```suggestion
const ResetPassword = ({ type }: { type?: 'add' }) => {
```

**Review by naumovski-filip — APPROVED** [review]

**Review by naumovski-filip — APPROVED** [review]

File: src/components/EditPasswordForm/EditPasswordForm.tsx
Comment (ChristiaanScheermeijer) [inline src/components/EditPasswordForm/EditPasswordForm.tsx]: This logic needs to be placed up one level (container). This change adds store logic to a component.

Comment (ChristiaanScheermeijer) [reply src/components/EditPasswordForm/EditPasswordForm.tsx]: You can then also combine the `errorMessage` with the form `errors` nicely.

File: src/components/EditPasswordForm/EditPasswordForm.tsx
Comment (ChristiaanScheermeijer) [inline src/components/EditPasswordForm/EditPasswordForm.tsx]: This condition seems a bit off to me. Shouldn't we make it explicit (via props) that a password will be added instead of edited? I'm also not against having a separate `<AddPasswordForm />` component if that will make it easier and will not be mostly duplicate code.

File: src/components/WarningAccountModal/WarningAccountModal.tsx
Comment (ChristiaanScheermeijer) [inline src/components/WarningAccountModal/WarningAccountModal.tsx:1]: `WarningAccountModal` sounds like it's a generic component to me, but it's actually really specific.

Here are my suggestions:

1. Rename it to `DeleteAccountWarning` or even more specific `DeleteAccountPasswordWarning`
2. It's also a component, so it should not contain any stores and preferably no handlers.

File: src/components/WarningAccountModal/WarningAccountModal.tsx
Comment (ChristiaanScheermeijer) [inline src/components/WarningAccountModal/WarningAccountModal.tsx]: I would also make the translation keys more specific:

```suggestion
      <h2 className={styles.heading}>{t('account.delete_password_warning_title')}</h2>
      <p className={styles.paragraph}>{t('account.delete_password_warning_text')}</p>
```

Or using an object:

```suggestion
      <h2 className={styles.heading}>{t('account.delete_password_warning.title')}</h2>
      <p className={styles.paragraph}>{t('account.delete_password_warning.text')}</p>
```

File: src/components/WarningAccountModal/WarningAccountModal.tsx
Comment (ChristiaanScheermeijer) [inline src/components/WarningAccountModal/WarningAccountModal.tsx]:
```suggestion
          label={t('account.delete_password_warning.proceed')}
```

**Comment by borkopetrovicc** [comment]
@ChristiaanScheermeijer I initially believed that you had already approved and commented 'LGTM'. Could we address these recent comments separately in another PR or ticket for improvements?

**Comment by kmitrovv** [comment]
> @ChristiaanScheermeijer I initially believed that you had already approved and commented 'LGTM'. Could we address these recent comments separately in another PR or ticket for improvements?

@borkopetrovicc the suggestions make sense to me and do not look like a lot of effort, so we better address them in this PR

**Comment by ChristiaanScheermeijer** [comment]
> @ChristiaanScheermeijer I initially believed that you had already approved and commented 'LGTM'. Could we address these recent comments separately in another PR or ticket for improvements?

Yes, sorry for that. I indeed approved after a quick scan, but after a second review (and some new changes), I have found new improvement points.

The suggested changes will not affect the outcome but will improve the code quality and testability of components. The changes shouldn't take long, so I agree with @kiremitrov123 to include them in this PR. But that's up to you :-)

**Comment by naumovski-filip** [comment]
@ChristiaanScheermeijer @AntonLantukh Because @borkopetrovicc is on sick leave I've taken a look at the most recent comments and tried to fix them. Regarding the failing E2E tests, there are issues on Cleeng's side.

File: src/components/EditPasswordForm/EditPasswordForm.tsx
Comment (ChristiaanScheermeijer) [inline src/components/EditPasswordForm/EditPasswordForm.tsx:48]: We should write this down for later, but I still have trouble understanding why this logic is in place. The edit password form implements two methods:

- Using your old password
- Using a password reset token received via mail

We assume a mix of these methods is possible, while that should not be the case. This can be solved by using a single prop, for example; `method: 'token' | 'old-password'`.

I assume that `showOldPasswordField` and `showResetTokenField` should never be `true` simultaneously. Is that correct? If so, I'm afraid the condition is still wrong.

File: public/locales/en/user.json
Comment (ChristiaanScheermeijer) [inline public/locales/en/user.json]: Oops! My bad, now it sounds like it's used for deleting your password 😅 

Last suggestion: `delete_account_password_warning`

File: public/locales/es/user.json
Comment (ChristiaanScheermeijer) [inline public/locales/es/user.json]: Do we want to keep the Spanish translations?

**Review by ChristiaanScheermeijer — COMMENTED** [review]
@naumovski-filip thank you for looking into this PR!

I think the only thing left is the resend email condition. The remaining feedback is not important for now (and partially my fault).

File: src/components/EditPasswordForm/EditPasswordForm.tsx
Comment (ChristiaanScheermeijer) [reply src/components/EditPasswordForm/EditPasswordForm.tsx:48]: As we discussed, the logic is correct but a bit weird, perhaps due to using existing props/variables for new use-cases.

We'll look into this later.

**Review by ChristiaanScheermeijer — APPROVED** [review]
Nice work! 🎉

**Pull request merged**
